src/core/agents.jl

```diff
@@ -196,7 +196,11 @@ macro agent(new_name, base_type, super_type, extra_fields)
     # This macro was generated with the guidance of @rdeits on Discourse:
     # https://discourse.julialang.org/t/
     # metaprogramming-obtain-actual-type-from-symbol-for-field-inheritance/84912
-
+    if base_type isa Expr
+        if base_type.args[1] == :ContinuousAgent && length(base_type.args) == 2
+            base_type = Expr(base_type.head, base_type.args..., :Float64)
+        end
+    end
```

Comment:
Okay, now I see. This is a hack that we should not do. It is unintuitive. by reading the usage code it was super confusing to me how this could happen, but it turns out to be a macro magic hack.

Of course, the problem is that without the hack pretty much every continuous space code will break. While now, pretty much nothing will break, except when a user explicitly compares agent positions to a hand-coded position `a.pos == (0.5, 0.2)`. 

So, the conclusion is: we leave this hack in, but we do not use it in the documentation. The docstring of `ContinuousAgent` explicitly mentions the **two** type parameters `D, T`. In all examples that use `ContinuousAgent` we explicitly inherit from `ContinuousAgent{D, Float64}` with appropriate `D`.

Additionally, we should add one more check in `do_checks` that the continous space extent type is also of type T.

Reply:
And, we add a comment just above line 199 in the macro that says "this is a hack for backwrards compatibilty, see PR 846."